qiskit/transpiler/__init__.py

```diff
@@ -34,84 +34,164 @@
-arguments (e.g. ``layout_method``). These can be set to one of the built-in methods and
-can also refer to available external plugins. See
-:mod:`qiskit.transpiler.preset_passmanagers.plugin` for details on this plugin interface.
-
-.. _working_with_preset_pass_managers:
-
-Working with Preset Pass Managers
-=================================
-
-Qiskit includes functions to build preset :class:`~.PassManager` objects.
-These preset passmanagers are used by the :func:`~.transpile` function
-for each optimization level. There are 4 optimization levels ranging from 0 to 3, where higher
-optimization levels take more time and computational effort but may yield a
-more optimal circuit.
-Optimization level 0 is intended for device characterization experiments and, as such, only
-maps the input circuit to the constraints of the target backend, without
-performing any optimizations. Optimization level 3 spends the most effort to optimize the circuit.
-However, as many of the optimization techniques in the transpiler are heuristic based, spending more
-computational effort does not always result in an improvement in the quality of the output
-circuit.
+Qiskit uses the graph-based :class:`.DAGCircuit` intermediate representation (IR) of a circuit
+throughout the transpiler stack, rather than the tree-based :class:`.QuantumCircuit`.  A transpiler
+pipeline is a :class:`.PassManager` object, whose :meth:`.PassManager.run` method takes in a
+:class:`.QuantumCircuit`, converts it to a :class:`.DAGCircuit`, then subjects the IR to a sequence
+of *passes*, finally returning a :class:`.QuantumCircuit` back.  A pass is either an
+:class:`.AnalysisPass`, whose purpose is to calculate and store properties about the circuit in the
+stateful :class:`.PropertySet`, or a :class:`.TransformationPass`, whose purpose is to modify the IR
+to achieve a particular singular goal.  We typically think of a pipeline as being split into
+"stages", where each stage is responsible for one high-level transformation.
+
+Qiskit exposes a default transpilation pipeline builder via the function
+:func:`.generate_preset_pass_manager`.  This will return a properly configured pipeline for complete
+transpilation, at a chosen ``optimization_level`` (between 0 and 3, inclusive).  Unless you are
+looking for something highly specialized, this is almost certainly the entry point you want.  A
+sample transpilation looks like::
+
+    from qiskit.circuit import QuantumCircuit
+    from qiskit.transpiler import generate_preset_pass_manager
+    from qiskit_ibm_runtime import QiskitRuntimeService
+
+    # Any abstract circuit you want:
+    abstract = QuantumCircuit(2)
+    abstract.h(0)
+    abstract.cx(0, 1)
+
+    # Any method you like to retrieve the backend you want to run on:
+    backend = QiskitRuntimeService().backend("some-backend")
+
+    # Create the pass manager for the transpilation ...
+    pm = generate_preset_pass_manager(backend=backend)
```

Comment:
We have had a few back and forths over the use of backend vs target as the "main" transpilation input, the original text said "target backend" so the terms where kind of interchangeable in the docs. Now, the code examples sometimes use `backend`, sometimes `target`, but the text mostly refers to the target. I don't think it's difficult to understand, but I wonder if we'd like to explain this nuance a bit better to users (maybe the answer is no).

Reply:
I don't have a strong preference with the state of Qiskit right this second, but I think Ian and I might be beginning to argue for an expansion of the meaning of `Backend` for a hypothetical `BackendV3` that will make it preferable to use `backend` as the default user-facing bit.

Reply:
I've left this as-is for now, but feel free to argue.